src/python/pants/core/util_rules/stripped_source_files.py

```diff
@@ -6,15 +6,26 @@
 
 from pants.core.util_rules.source_files import SourceFiles
 from pants.core.util_rules.source_files import rules as source_files_rules
+from pants.engine.collection import Collection
 from pants.engine.fs import Digest, DigestSubset, MergeDigests, PathGlobs, RemovePrefix, Snapshot
 from pants.engine.rules import Get, MultiGet, collect_rules, rule
-from pants.source.source_root import SourceRootsRequest, SourceRootsResult
+from pants.engine.target import SourcesPaths
+from pants.source.source_root import (
+    SourceRoot,
+    SourceRootRequest,
+    SourceRootsRequest,
+    SourceRootsResult,
+)
 from pants.source.source_root import rules as source_root_rules
+from pants.util.dirutil import fast_relpath
 
 
 @dataclass(frozen=True)
 class StrippedSourceFiles:
-    """Wrapper for a snapshot of files whose source roots have been stripped."""
+    """Wrapper for a snapshot of files whose source roots have been stripped.
+
+    Use via `Get(StrippedSourceFiles, SourceFilesRequest([tgt.get(Sources)])`.
```

Comment:
One of the biggest points of feedback about the Rules API is not knowing the magic incantation of what to put in the `Get`. We should get in the habit of doing this imo. It'll be really helpful when we have generated docs online.

Reply:
This kind of information is available in the rule graph (and rendered by `--native-engine-visualize-to`)... rather than adding it manually to comments, generating docs from the rule graph might be less error prone.

Reply:
Fair, but we've gotten feedback that `--native-engine-visualize-to` is too much info at the moment. Ack that this risks going stale, but until we have that*, I think this is better than the status quo.

*One idea I still really like is to output a graph with only the types, like shown in https://www.pantsbuild.org/v2.1/docs/rules-api-concepts#the-rule-graph.